Upgrade nudge: Update copy

sirreal opened the pull request:
- Use the long plan name (`WordPress.com Premium`/`Jetpack Premium`) instead of short (`Premium`/`Premium`) in the upgrade nudge. Requested by @dereksmart: p1563283534127100-slack-jetpack-crew
- Lower case "plan" in `WordPress.com Premium plan` and `Jetpack Premium plan`. I recall this guideline but can't find the reference right now.

## Screens

### Before

![Screen Shot 2019-07-16 at 16 17 53](https://user-images.githubusercontent.com/841763/61302267-c53d7780-a7e5-11e9-81d5-ead8f1ce89e7.png)

### After

![Screen Shot 2019-07-16 at 16 17 39](https://user-images.githubusercontent.com/841763/61302274-c8d0fe80-a7e5-11e9-9a16-57dbcb07fde8.png)

## Testing
- You'll need to set the constant `define( 'JETPACK_SHOW_BLOCK_UPGRADE_NUDGE', true );`
- With a site on a free plan, add the Simple Payments block.
- Make sure the nudge displays as expected (see screenshots).

sirreal commented:
cc: @mtias and @shaunandrews for design feedback on the long plan names.

ockham commented:
I don't think we ever refer to a WP.com plan by its full name (e.g. "WordPress.com Premium") on WP.com, which is why I opted to use the short name ("Premium").

shaunandrews commented:
I'm not aware of any previous decisions around how we refer to our plan names. However, I always appreciate brevity in our copy. The addition of "WordPress.com" or "Jetpack" doesn't seem to be super helpful.

--

Perhaps unrelated to this PR specifically, I do have a problem with the rest of the sentence. :)

> This block is available...

The first part of the message is misleading. We should be more forceful here and get to the point quicker. Something like:

> This block is disabled. Upgrade to the Premium plan to enable it.
> Site visitors will not see this block until you upgrade your plan.

Or...

> This block is not available with our current plan.
> This block will be hidden from view unless you upgrade to Premium.

jancavan commented:
I don't have a strong preference about adding "WordPress.com" or "Jetpack", but I do think the addition of "plan" is not necessary.

A couple additional comments:

1. I think we use the `banner` component for upgrade nudges: https://wpcalypso.wordpress.com/devdocs/design/banner. I don't see the left border on the screenshots.

2. I agree with Shaun about being more direct in the copy. Another suggestion I'd like to make:

> This block is not available with your current plan.
> Your site visitors will not see this block until you upgrade to Premium.

3. I'm assuming this PR covers all upgrade nudges within the product?

![Screenshot 2019-07-16 08 33 31](https://user-images.githubusercontent.com/4924246/61308162-80ddb780-a7a4-11e9-9cfd-eeb84314a992.png)

sirreal commented:
Thanks for the feedback, folks. I'll propose a different solution soon taking into account your feedback.

> 1. I think we use the `banner` component for upgrade nudges: https://wpcalypso.wordpress.com/devdocs/design/banner. I don't see the left border on the screenshots.

This nudge is only in the context of the Gutenberg editor and is built from the [Gutenberg `Warning` component](https://github.com/WordPress/gutenberg/tree/master/packages/block-editor/src/components/warning). We can work to style this component more like the `Banner`, but we can't use it directly because this is outside of Calypso. We have been hoping to invest efforts into making components shareable, but nothing is ready yet.

> 2. I agree with Shaun about being more direct in the copy. Another suggestion I'd like to make:
> 
> > This block is not available with your current plan.
> > Your site visitors will not see this block until you upgrade to Premium.

👍 I'll prepare one of your suggestions in another PR.

> 3. I'm assuming this PR covers all upgrade nudges within the product?
> 
> ![Screenshot 2019-07-16 08 33 31](https://user-images.githubusercontent.com/4924246/61308162-80ddb780-a7a4-11e9-9cfd-eeb84314a992.png)

No, this is only for the new nudges introduced for blocks that require paid plans. There are more details here: p58i-7Tf-p2

sirreal commented:
> We can work to style this component more like the `Banner`

@shaunandrews already brought this up, see p58i-7Tf-p2#comment-42390 and p58i-7Tf-p2#comment-42417

Before iterating on `Banner`-like styling, I'll wait for some resolution there.

sirreal committed:
[not verified] Update copy per suggestions

sirreal committed:
[not verified] Use long product name

ranh commented:
The brand book has this to say about plan names: p9GxdL-h5-p2

> In general, refer to plans by their full names the first time they’re mentioned on a page or in an email.
The full name is the product name followed by the specific descriptor:
>  
> > - WordPress.com Personal 
> > - WordPress.com Premium 
> > - WordPress.com Business
> > - Jetpack Personal 
> > - Jetpack Premium 
> > - Jetpack Professional
>
> After the first mention, use just the descriptor plus the word “plan.” The word “plan” isn’t part of the name, so it doesn’t get capitalized:
>
> > If WordPress.com Premium isn’t enough, consider our Business plan.

So "plan" shouldn't be capitalized, and at least according to the brand book, we should use the full plan name anyway, regardless of the potential confusion between wpcom and JP (which I understand to be the reason behind @dereksmart's suggestion)

That said, I'm not sure this should apply to in-app screens. I think this is more applicable to "static" pages like articles, landing pages, emails. etc – where we would mention plan names multiple times. Inside a notification or a notice, I think reducing the word count is much more important. 

It's worth digging more into this potential confusion. For example, is it important to clear it up in this notice, or can it be cleared up later on during the purchase flow?

I also agree with Shaun's feedback, we should send a more positive message and open with the most important information:

> **Upgrade to (plan) to use this block on your site.**
> You can try it out before upgrading, but only you will see it. It will be hidden from visitors until you upgrade.

I also wanted to try to better promote the plan upgrade:

> **Upgrade to (plan) to use this block, and get access to (many more cool features)**

"Many more cool features" could be contextual, and include details about the specific plan suggested.

ockham commented:
Thanks @ranh for looking this up in the brand book!

As for confusion between WP.com and JP plans -- I didn't really see any potential for confusion there, since the respective upgrade nudges will always only show up in the editor on _either_ a WP.com site, _or_ a Jetpack site, which for me seemed like sufficient context for the user.

sirreal commented:
Thanks @ranh!

> It's worth digging more into this potential confusion.

I'd highlight that this is in the context of the block editor. There's potential in this context for a variety of blocks from different providers to offer upgrades. The confusion may not be "is this WordPress.com or Jetpack" but more generally "Who is trying to sell me this _Premium_ upgrade?"

---

There are a couple of issues with the layout of the block I'm working to clear up, then it would be good to settle on the definitive copy.

ranh commented:
> I'd highlight that this is in the context of the block editor. There's potential in this context for a variety of blocks from different providers to offer upgrades. The confusion may not be "is this WordPress.com or Jetpack" but more generally "Who is trying to sell me this Premium upgrade?"

That's not really clear, you mean that this notice will be used to show upgrade prompts for arbitrary blocks, not just for wpcom and JP?

Unless that's the case, if we don't need to specifically distinguish between wpcom and JP plans, I don't see a reason to include the product name here.

sirreal commented:
> That's not really clear, you mean that this notice will be used to show upgrade prompts for arbitrary blocks, not just for wpcom and JP?

No, at the moment the only blocks we plan to use this nudge for are blocks included in WordPress.com and Jetpack plans.

What I mean is that on Jetpack sites and some WordPress.com sites, there may be other blocks in the editor from other providers trying to sell upgrades. We don't know. The editor is a level playing field. If the "Jetpack" and "WordPress.com" parts of the plan names are removed, nothing indicates what sort of thing you're upgrading or where the block even comes from:

![Screen Shot 2019-07-17 at 11 38 27](https://user-images.githubusercontent.com/841763/61365239-ad1e3480-a887-11e9-826c-b8a4637bd528.png)

It's true that it should be clear as soon as you click upgrade and enter the checkout flow:

![Screen Shot 2019-07-17 at 11 43 03](https://user-images.githubusercontent.com/841763/61365531-32a1e480-a888-11e9-8c07-bc5686205a67.png)

sirreal commented:
This PR got away from me a bit, I'd like to keep it focused on the copy. Here is the current copy for review (updated in description):

> This block is not available with your current plan.
> Your site visitors will not see this block until you upgrade to Jetpack Premium.

![Screen Shot 2019-07-17 at 14 14 40](https://user-images.githubusercontent.com/841763/61374706-53286980-a89d-11e9-90f1-39ab720abfb9.png)

On WordPress.com, the plan name will read "[…] upgrade to WordPress.com Premium."

shaunandrews commented:
Did you move the button down to accommodate the longer text? I don't mind it, just curious.

Perhaps we could align the left edge of the button with the left edge of the text?

Not to throw another wrench in this, but how does it work when its nested within another block?

sirreal commented:
> Did you move the button down to accommodate the longer text? I don't mind it, just curious.

I did not do anything special, this is the `Warning` flex layout worked with longer text.

> Perhaps we could align the left edge of the button with the left edge of the text?

Here's what that would look like:

<img width="530" alt="Screen Shot 2019-07-17 at 23 24 57" src="https://user-images.githubusercontent.com/841763/61412822-264e7380-a8ea-11e9-8fd0-b41a54c696d7.png">

Or inline depending on the text length:

<img width="530" alt="Screen Shot 2019-07-17 at 23 26 03" src="https://user-images.githubusercontent.com/841763/61412868-54cc4e80-a8ea-11e9-853a-06e2c5fca4f9.png">


> Not to throw another wrench in this, but how does it work when its nested within another block?

Is there a specific behavior you're wondering about? To work with more blocks, there's some logic that needs to be changed in PHP where the blocks are registered then they'll be rendered with the nudge. Here's an example with Related Posts (just a random block I picked for demonstration):

![related-posts](https://user-images.githubusercontent.com/841763/61413542-1172df80-a8ec-11e9-84b7-f380afb5f0e7.png)

ranh commented:
@sirreal can you try this with the copy suggested here? https://github.com/Automattic/jetpack/pull/13065#issuecomment-512169502

sirreal committed:
Use suggested copy

> Upgrade to (plan) to use this block on your site.
> You can try it out before upgrading, but only you will see it. It will be hidden from visitors until you upgrade.

via https://github.com/Automattic/jetpack/pull/13065#issuecomment-512169502

sirreal commented:
@ranh Here's what that looks like (I've pushed a commit with this):

![Screen Shot 2019-07-18 at 09 43 03](https://user-images.githubusercontent.com/841763/61438775-85d76e00-a940-11e9-88c0-61ec285fad63.png)

sirreal commented:
The loading state is related, as long as I've got lots of design eyes, would be great to look at that as well 😉 🙇 

https://github.com/Automattic/jetpack/pull/13054

jeherve reviewed (changes requested):
This is not super pretty in IE11 right now:

![image](https://user-images.githubusercontent.com/426388/61453910-a7941d80-a95f-11e9-8c91-1a64ed76f62a.png)

Would there be a way to wrap the text in those conditions?

sirreal commented:
> This is not super pretty in IE11 right now 
> Would there be a way to wrap the text in those conditions?

@jeherve I suspect that's an existing issue that's highlighted by the longer text here. I'll see about addressing it in another PR.
